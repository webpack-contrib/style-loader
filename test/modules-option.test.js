import path from 'path';

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  runInJsDom,
} from './helpers/index';

describe('"modules" option', () => {
  const injectTypes = [
    'styleTag',
    'singletonStyleTag',
    'lazyStyleTag',
    'lazySingletonStyleTag',
  ];

  injectTypes.forEach((injectType) => {
    it(`should work with the "${injectType}" inject type`, async () => {
      expect.assertions(3);

      const compiler = getCompiler(
        './named-export.js',
        {},
        {
          module: {
            rules: [
              {
                test: /\.css$/i,
                use: [
                  {
                    loader: path.resolve(__dirname, '../src/cjs.js'),
                    options: {
                      injectType,
                      esModule: true,
                      modules: { namedExport: true },
                    },
                  },
                  {
                    loader: 'css-loader',
                    options: {
                      modules: {
                        namedExport: true,
                      },
                    },
                  },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });
  });
});
