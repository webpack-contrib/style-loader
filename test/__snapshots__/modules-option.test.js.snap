// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"modules" option should work with the "lazySingletonStyleTag" inject type: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


<div class=\\"KDvN8WTqb8V1KZHXZP-v\\">Water</div><div class=\\"_2ytpV86orKhgjYu_KpCdi _2SqzNMl3NVNNilgJp7bbug\\">Ground</div></body></html>"
`;

exports[`"modules" option should work with the "lazySingletonStyleTag" inject type: errors 1`] = `Array []`;

exports[`"modules" option should work with the "lazySingletonStyleTag" inject type: warnings 1`] = `Array []`;

exports[`"modules" option should work with the "lazyStyleTag" inject type: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


<div class=\\"KDvN8WTqb8V1KZHXZP-v\\">Water</div><div class=\\"_2ytpV86orKhgjYu_KpCdi _2SqzNMl3NVNNilgJp7bbug\\">Ground</div></body></html>"
`;

exports[`"modules" option should work with the "lazyStyleTag" inject type: errors 1`] = `Array []`;

exports[`"modules" option should work with the "lazyStyleTag" inject type: warnings 1`] = `Array []`;

exports[`"modules" option should work with the "singletonStyleTag" inject type: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style>._2SqzNMl3NVNNilgJp7bbug {
  background: red;
}
.KDvN8WTqb8V1KZHXZP-v {
    background: red;
}

._2ytpV86orKhgjYu_KpCdi {
    color: blue;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


<div class=\\"KDvN8WTqb8V1KZHXZP-v\\">Water</div><div class=\\"_2ytpV86orKhgjYu_KpCdi _2SqzNMl3NVNNilgJp7bbug\\">Ground</div></body></html>"
`;

exports[`"modules" option should work with the "singletonStyleTag" inject type: errors 1`] = `Array []`;

exports[`"modules" option should work with the "singletonStyleTag" inject type: warnings 1`] = `Array []`;

exports[`"modules" option should work with the "styleTag" inject type: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style>._2SqzNMl3NVNNilgJp7bbug {
  background: red;
}
</style><style>.KDvN8WTqb8V1KZHXZP-v {
    background: red;
}

._2ytpV86orKhgjYu_KpCdi {
    color: blue;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


<div class=\\"KDvN8WTqb8V1KZHXZP-v\\">Water</div><div class=\\"_2ytpV86orKhgjYu_KpCdi _2SqzNMl3NVNNilgJp7bbug\\">Ground</div></body></html>"
`;

exports[`"modules" option should work with the "styleTag" inject type: errors 1`] = `Array []`;

exports[`"modules" option should work with the "styleTag" inject type: warnings 1`] = `Array []`;
